src/lib/server/auth/index.ts

```diff
@@ -1,22 +1,43 @@
 import { redirect, type ServerLoadEvent } from '@sveltejs/kit';
-import dayjs from 'dayjs';
 import { prisma } from '../prisma';
+import { createClerkClient } from '@clerk/express';
+import { CLERK_SECRET_KEY } from '$env/static/private';
+import { clerkClient } from 'clerk-sveltekit/server';
 
-export async function validateSession(event: ServerLoadEvent) {
-	const sessionId = event.cookies.get('auth_session');
-	if (!sessionId) {
-		redirect(302, '/login');
+const clerkSessionClient = createClerkClient({
+	secretKey: CLERK_SECRET_KEY,
+});
+
+export async function validateSession({ locals }: ServerLoadEvent) {
+	if (!locals.auth.userId) {
+		return redirect(307, '/login');
 	}
-	const session = await prisma.session.findUnique({
-		where: { id: sessionId },
-		include: { user: true },
-	});
-	if (!session || !session.user) {
-		redirect(302, '/login');
+
+	if (!locals.auth.orgId && locals.auth.sessionId) {
+		// Sign out the user if they are not associated with an organization
+		await clerkSessionClient.sessions.revokeSession(locals.auth.sessionId);
+		return redirect(307, '/login');
 	}
-	const expiry = session.expiresAt;
-	if (dayjs(expiry).isBefore(dayjs())) {
-		redirect(302, '/login');
+
+	const clerkUser = await clerkClient.users.getUser(locals.auth.userId);
+
+	let user = await prisma.user.findFirst({
+		where: {
+			clerkId: clerkUser.id,
+		},
+	});
+
+	if (!user) {
+		user = await prisma.user.create({
+			data: {
+				clerkId: clerkUser.id,
+				email: clerkUser.emailAddresses[0].emailAddress,
+				name: clerkUser.fullName ?? 'Ben the Man',
```

Comment:
???

Reply:
don't like that 😉 ?

Reply:
WE ARE PROFESSIONALS HERE MOE

Reply:
I will change it to an empty string, I thought we were fun professionals

Reply:
is the user fullName optional...? I'd rather not default to a magic string

Reply:
If there is a situation where name will be empty/undefined consider changing the db schema